src/vma/dev/ring_allocation_logic.cpp

```diff
@@ -182,6 +190,14 @@ bool ring_allocation_logic::should_migrate_ring()
 	return true;
 }
 
+const char* ring_allocation_logic::to_str()
+{
+	if (unlikely(m_str[0] == '\0')) {
+		snprintf(m_str, sizeof(m_str), "[%s=%p]", m_type, m_owner);
```

Comment:
Please consider adding return error check if you see any potential issues. Static analysis tools as BlackDuck can report security issues.

Reply:
Please, check the last commit. Does it do what you meant?